pghoard/basebackup.py

```diff
@@ -97,20 +97,27 @@ def compress_directly_to_a_file(self, proc, basebackup_path):
         encryption_key_id = self.config["backup_sites"][self.site]["encryption_key_id"]
         if encryption_key_id:
             rsa_public_key = self.config["backup_sites"][self.site]["encryption_keys"][encryption_key_id]["public"]
-        c = Compressor()
+
         compression_algorithm = self.config["compression"]["algorithm"]
         compression_level = self.config["compression"]["level"]
         self.log.debug("Compressing basebackup directly to file: %r", basebackup_path)
         set_stream_nonblocking(proc.stderr)
 
         with NamedTemporaryFile(prefix=basebackup_path, suffix=".tmp-compress") as output_obj:
-            original_input_size, compressed_file_size = c.compress_fileobj(
+            def progress_callback():
+                stderr_data = proc.stderr.read()
```

Comment:
How does this blocking read affect the stdout data processing in the rohmufile.write_file()? I assume that since --verbose, it chunks the stdout reads somewhat. Performance impact? Would it be better to block only in the stdout read?